app/main.py

```
@@ -1,8 +1,30 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+
+    def __init__(self, name , age):
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    Person.people.clear()
+    person_objects  = [Person(p["name"], p["age"]) for p in people]
+
+    for person_dict , person_object in zip(people, person_objects ):
+        partner_wife = person_dict.get("wife")
+        partner_husband = person_dict.get("husband")
+
+
+        if partner_wife and partner_wife in Person.people:
+            person_object.wife = Person.people[partner_wife]
+        else:
+            raise ValueError(f"Person with name '{partner_wife}' not found")
```

This `if/else` structure incorrectly handles cases where a person has no wife. If `partner_wife` is `None`, the condition is false and the `else` block is executed, raising a `ValueError`. The code should only raise an error if a wife's name is given but not found in `Person.people`.